graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndicesAdapterES7.java

```diff
@@ -417,8 +422,13 @@ public void optimizeIndex(String index, int maxNumSegments, Duration timeout) {
                 .indices(index)
                 .maxNumSegments(maxNumSegments)
                 .flush(true);
-
-        client.execute((c, requestOptions) -> c.indices().forcemerge(request, withTimeout(requestOptions, timeout)));
+        try {
+            final FutureActionListener<ForceMergeResponse> future = new FutureActionListener<>();
+            client.execute((c, requestOptions) -> c.indices().forcemergeAsync(request, withTimeout(requestOptions, timeout), future));
+            future.getBlocking(timeout);
```

Comment:
I'd say this code block should be given a comment.
One may wonder what is the point of introducing an async call if in the next line you block the execution with getBlocking()? I believe you should state clearly that it is a trick and you hope to save threads from connection pool with this trick.